crates/forge/src/scarb.rs

```diff
@@ -36,23 +41,70 @@ pub fn build_contracts_with_scarb(filter: PackagesFilter) -> Result<()> {
 
 pub fn build_test_artifacts_with_scarb(filter: PackagesFilter) -> Result<()> {
     ScarbCommand::new_with_stdio()
-        .arg("snforge-test-collector")
+        .arg("build")
+        .arg("--test")
         .packages_filter(filter)
         .run()
         .context("Failed to build test artifacts with Scarb")?;
     Ok(())
 }
 
+/// collecting by name allow us to dedup targets
+/// we do it because they use same sierra and we display them without distinction anyway
+fn test_targets_by_name(package: &PackageMetadata) -> HashMap<String, &TargetMetadata> {
```

Comment:
So we group targets by name. But in what cases can the target be duplicated if we haven't done this?

Reply:
For integration tests Scarb marks compilation units and targets by group-id, CUs with the same group-id are compiled together. This is done to optimize the compilation (the virtual `lib.cairo` trick I did ~a year ago in our test collector). Simple example: a package with `tests.cairo` and `test_contract.cairo` files in `tests` folder

```json  
      {
          "kind": "test",
          "name": "hello_world_test",
          "source_path": ".../uwu/tests/test.cairo",
          "params": {
            "group-id": "hello_world_integrationtest",
            "test-type": "integration"
          }
        },
        {
          "kind": "test",
          "name": "hello_world_test_contract",
          "source_path": ".../uwu/tests/test_contract.cairo",
          "params": {
            "group-id": "hello_world_integrationtest",
            "test-type": "integration"
          }
        },
```